pkgs/unified_analytics/lib/src/initializer.dart

```diff
@@ -75,17 +73,12 @@ class Initializer {
     logFile.createSync(recursive: true);
   }
 
-  /// Creates the session json file which will contain
-  /// the current session id along with the timestamp for
-  /// the last ping which will be used to increment the session
-  /// if current timestamp is greater than the session window.
+  /// Creates the session file which will contain
+  /// the current session id which is the current timestamp.
   static void createSessionFile({required File sessionFile}) {
     final now = clock.now();
     sessionFile.createSync(recursive: true);
-    sessionFile.writeAsStringSync(jsonEncode(<String, int>{
-      'session_id': now.millisecondsSinceEpoch,
-      'last_ping': now.millisecondsSinceEpoch,
-    }));
+    sessionFile.writeAsStringSync('${now.millisecondsSinceEpoch}');
```

Comment:
Actually, when we discussed this yesterday, I was assuming this was the *client* ID, and we'd never need to re-write this. Since this is the *session* ID, this means we would still need to write to this when starting a new session, and another process could read from it in the middle of a write, do I have that correct?

If so, I have a few thoughts:

1. using last modified timestamp still seems like an improvement, and we'd at least get less race conditions
2. STILL using JSON is probably better, because then we would get a parse error vs. just getting the WRONG session_id
3. let's just hard-code the json as a string, rather than invoking the overhead of `jsonEncode`
4. maybe it's enough to catch the parse error, and then re-try 1 second later?

Reply:
I agree with all of these points. Removing the last ping key should be easy enough. Also with the refactor to prevent a huge amount of write operations, I think we may be able to fall back to attempting to `paraeContents` again?

If we see that this is still coming up in the error logs for the race condition, we can return to setting the sessionId back to the current timestamp. Thoughts?

Reply:
> I think we may be able to fall back to attempting to paraeContents again?

It would still be possible for the second parse to fail, right?

> If we see that this is still coming up in the error logs for the race condition, we can return to setting the sessionId back to the current timestamp.

Not sure what you mean here. Are you proposing using something else other than the current timestamp for the sessionId?

Reply:
> It would still be possible for the second parse to fail, right?

Yes, the `parseContents` method always has the possibility of failing.

> Not sure what you mean here. Are you proposing using something else other than the current timestamp for the sessionId?

And what I meant here is monitoring the crash logs to see if we run into this race condition again... i would suspect that we would have a lot less instances of this error since we are now writing a lot less by checking `File.lastModifiedSync` method